client/components/search/autocompleters/product.js

```diff
@@ -0,0 +1,80 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { stringify } from 'qs';
+
+/**
+ * Internal dependencies
+ */
+import ProductImage from 'components/product-image';
+
+const computeSuggestionMatch = ( suggestion, query ) => {
+	if ( ! query ) {
+		return null;
+	}
+	const indexOfMatch = suggestion.toLocaleLowerCase().indexOf( query.toLocaleLowerCase() );
+
+	return {
+		suggestionBeforeMatch: suggestion.substring( 0, indexOfMatch ),
+		suggestionMatch: suggestion.substring( indexOfMatch, indexOfMatch + query.length ),
+		suggestionAfterMatch: suggestion.substring( indexOfMatch + query.length ),
+	};
+};
+
+/**
+ * A products completer.
+ * See https://github.com/WordPress/gutenberg/tree/master/packages/components/src/autocomplete#the-completer-interface
+ *
+ * @type {Completer}
+ */
+export default {
+	name: 'products',
+	className: 'woocommerce-search__product-result',
+	options( search ) {
+		let payload = '';
+		if ( search ) {
+			const query = {
+				search: encodeURIComponent( search ),
+				per_page: 10,
+				orderby: 'popularity',
+			};
+			payload = '?' + stringify( query );
+		}
+		return apiFetch( { path: '/wc/v3/products' + payload } );
```

Comment:
I'm using `apiFetch` directly since I don't think these search term results need to be cached - component state works well enough to find & display the results.

Reply:
Agree here